main.go

```diff
@@ -54,20 +70,28 @@ func readConfig(cfg *proxy.Config) {
 	}
 
 	// Perform custom overrides.
-	if primePort != nil && *primePort != -1 {
-		cfg.Upstream[common.PRIME_CTX].Url = "ws://0.0.0.0:" + strconv.Itoa(*primePort)
+	if primePort != nil && *primePort != "prime" {
+		cfg.Upstream[common.PRIME_CTX].Url = "ws://127.0.0.1:" + returnPortHelper(*primePort)
 	}
-	if regionPort != nil && *regionPort != -1 {
-		cfg.Upstream[common.REGION_CTX].Url = "ws://0.0.0.0:" + strconv.Itoa(*regionPort)
+	if regionPort != nil && *regionPort != "" {
+		cfg.Upstream[common.REGION_CTX].Url = "ws://127.0.0.1:" + returnPortHelper(*regionPort)
 	}
-	if zonePort != nil && *zonePort != -1 {
-		cfg.Upstream[common.ZONE_CTX].Url = "ws://0.0.0.0:" + strconv.Itoa(*zonePort)
+	if zonePort != nil && *zonePort != "" {
+		cfg.Upstream[common.ZONE_CTX].Url = "ws://127.0.0.1:" + returnPortHelper(*zonePort)
 	}
 	if *stratumPort != -1 {
 		cfg.Proxy.Stratum.Listen = "0.0.0.0:" + strconv.Itoa(*stratumPort)
 	}
 }
 
+func returnPortHelper(portStr string) string {
```

Comment:
Thanks for this!

Reply:
Yeah great idea

Reply:
The README is pretty out of date, so its an entire rewrite